apps/cli/src/runAstgrepCodemod.ts

```diff
@@ -0,0 +1,62 @@
+import fs from 'fs';
+import { execSync } from 'node:child_process';
+import { readFile } from 'node:fs/promises';
+import path from 'path';
+import * as yaml from 'js-yaml';
+import { PrinterBlueprint } from './printer.js';
+
+export const runAstgrep = async (
+	printer: PrinterBlueprint,
+	rulePath: string,
+	targetDirectory: string,
+): Promise<void> => {
+	const yamlString = await readFile(rulePath, { encoding: 'utf8' });
+	const yamlObject = yaml.load(yamlString);
+	const language = languageToExtension(yamlObject.language);
+	printer.printConsoleMessage(
+		'info',
+		`Executing ast-grep for language : ${language}`,
+	);
+
+	// Function to recursively iterate over files in the directory
+	const iterateFiles = async (dirPath: string) => {
+		const entries = await fs.promises.readdir(dirPath, {
```

Comment:
do we need to take into account `include` and `exclude` patterns here (passed as cli args)?

Reply:
maybe we can use `glob` lib for this use case?

Reply:
I was thinking of shipping this and doing other changes once Sasha refactors the cli.